src/OpenTelemetry/BaseExporter.cs

```diff
@@ -105,6 +106,14 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
+        public virtual AggregationTemporality GetAggregationTemporality()
```

Comment:
Could we do `BaseMetricExporter : BaseExporter<Metric>` and define GetAggregationTemporality() inside BaseMetricExporter only?

Reply:
Probably just call it `MetricExporter`?

Reply:
Yes, so I tried this out originally, the stickiness I ran into is that we'd like to be able to do `ConsoleMetricExporter : ConsoleExporter<Metric>` same kind of thing for `OtlpMetricExporter : OltpExporter<Metric>`. The base console and OTLP exporters contain common logic we're hoping to take advantage of.

Though, we might be fighting the type hierarchy too hard. It may make more sense to move this common logic to separate classes and have a handle to it. This would enable us to do something like `OtlpMetricExporter : MetricExporter : BaseExporter<Metric>`.

I plan to experiment with this in a follow up.